geopy/geocoders/here.py

```diff
@@ -0,0 +1,281 @@
+
+        :param dict other: A dictionary with additional key/values as described
+            in the online documentation. These will be added as individual query
+            parameters to the URL.
+
+        :rtype: ``None``, :class:`geopy.location.Location` or a list of them, if
+            ``exactly_one=False``.
+        """
+        point = self._coerce_point_to_string(query)
+        params = {
+            'app_id': self.app_id,
+            'app_code': self.app_code,
+            'maxresults': 1,
+            'mode': 'retrieveAddresses',  # can be overwritten using `other`
+            'prox': point + ',1000',
+        }
+        if exactly_one:
+            params['maxresults'] = 1
+        if language:
+            params['language'] = language
+        if other:
+            params.update(other)
+
+        url = "%s?%s" % (
+            self.reverse_api, urlencode(params))
+        logger.debug("%s.reverse: %s", self.__class__.__name__, url)
+        return self._parse_json(
+            self._call_geocoder(url, timeout=timeout),
+            exactly_one
+        )
+
+    @staticmethod
+    def _parse_json(doc, exactly_one=True):
+        """
+        Parse a location name, latitude, and longitude from an JSON response.
+        """
+        status_code = doc.get("statusCode", 200)
+        if status_code != 200:
+            err = doc.get("errorDetails", "")
+            if status_code == 401:
+                raise GeocoderAuthenticationFailure(err)
+            elif status_code == 403:
+                raise GeocoderInsufficientPrivileges(err)
+            elif status_code == 429:
+                raise GeocoderQuotaExceeded(err)
+            elif status_code == 503:
+                raise GeocoderUnavailable(err)
+            else:
+                raise GeocoderServiceError(err)
+
+        try:
+            resources = doc['Response']['View'][0]['Result']
+        except IndexError:
+            resources = None
+        if resources is None or not len(resources):
+            return None
+
+        def parse_resource(resource):
+            """
+            Parse each return object.
+            """
+            stripchars = ", \n"
+            addr = resource['Location']['Address']
+
+            address = addr.get('Label', '').strip(stripchars)
+            city = addr.get('City', '').strip(stripchars)
+            state = addr.get('State', '').strip(stripchars)
+            zipcode = addr.get('PostalCode', '').strip(stripchars)
+            country = addr.get('Country', '').strip(stripchars)
+
+            city_state = join_filter(", ", [city, state])
+            place = join_filter(" ", [city_state, zipcode])
+            location = join_filter(", ", [address, place, country])
+
+            display_pos = resource['Location']['DisplayPosition']
+            latitude = display_pos['Latitude'] or None
+            longitude = display_pos['Longitude'] or None
+            if latitude and longitude:
```

Comment:
Does the documentation say that these might be missing? Sorry I didn't look it myself.

If it doesn't, maybe we could assume that it's always there and fail when it's not? For example, `float(None)` would raise a TypeError exception. So instead we can simply do this:
```
latitude = float(display_pos['Latitude'])
longitude = float(display_pos['Longitude'])
```

Also the `or None` part would eat up `0.0`.